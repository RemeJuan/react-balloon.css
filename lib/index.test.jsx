import React from 'react';
import { create as rendererCreate } from 'react-test-renderer';

import Tooltip from './';

describe('<Tooltip />', () => {
  it('should render with default props', () => {
    const props = {};

    const wrapper = rendererCreate(
      <Tooltip {...props} />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with message', () => {
    const props = {
      message: 'Test message',
    };

    const wrapper = rendererCreate(
      <Tooltip {...props} />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with length', () => {
    const props = {
      length: 'small',
    };

    const wrapper = rendererCreate(
      <Tooltip {...props} />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with blockChild', () => {
    const props = {
      blockChild: false,
    };

    const wrapper = rendererCreate(
      <Tooltip {...props} />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with always', () => {
    const props = {
      always: true,
    };

    const wrapper = rendererCreate(
      <Tooltip {...props} />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with position: left', () => {
    const props = {
      position: 'left',
    };

    const wrapper = rendererCreate(
      <Tooltip {...props} />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with position: right', () => {
    const props = {
      position: 'right',
    };

    const wrapper = rendererCreate(
      <Tooltip {...props} />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with position: bottom', () => {
    const props = {
      position: 'bottom',
    };

    const wrapper = rendererCreate(
      <Tooltip {...props} />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
