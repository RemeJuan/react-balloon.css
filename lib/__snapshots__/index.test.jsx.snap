// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render with always 1`] = `
<span
  className="sc-bdVaJa khkMdF"
/>
`;

exports[`<Tooltip /> should render with blockChild 1`] = `
<span
  className="sc-bdVaJa iVAwZw"
/>
`;

exports[`<Tooltip /> should render with default props 1`] = `
<span
  className="sc-bdVaJa iVAwZw"
/>
`;

exports[`<Tooltip /> should render with length 1`] = `
<span
  className="sc-bdVaJa fZSyEg"
/>
`;

exports[`<Tooltip /> should render with message 1`] = `
<span
  className="sc-bdVaJa eDmcWv"
/>
`;

exports[`<Tooltip /> should render with position: bottom 1`] = `
<span
  className="sc-bdVaJa fAqnua"
/>
`;

exports[`<Tooltip /> should render with position: left 1`] = `
<span
  className="sc-bdVaJa jCuedB"
/>
`;

exports[`<Tooltip /> should render with position: right 1`] = `
<span
  className="sc-bdVaJa hjPhfO"
/>
`;
